packages/common/pitop/common/ptdm.py

```diff
@@ -457,35 +459,37 @@ def __thread_method(self):
         poller.register(self._zmq_socket, zmq.POLLIN)
         while self.__continue:
             events = poller.poll(_TIMEOUT_MS)
-
             for _ in range(len(events)):
                 message_string = self._zmq_socket.recv_string()
                 message = Message.from_string(message_string)
 
-                id = message.message_id()
-                if id in self._callback_funcs:
-                    self.invoke_callback_func_if_exists(
-                        self._callback_funcs[id], message.parameters
-                    )
-
-    def invoke_callback_func_if_exists(self, func, params=list()):
-        if not callable(func):
-            return
-
-        func_arg_no = len(signature(func).parameters)
-        if func_arg_no > 1:
-            logger.error(
-                "Invalid callback function - it should receive at most one argument."
-            )
-            return ""
+                callback = self._callback_funcs.get(message.message_id())
+                if callback:
+                    self.invoke_callback_func_if_exists(callback, message.parameters)
 
+    def invoke_callback_func_if_exists(self, callback, params=list()):
+        func_arg_no = len(signature(callback).parameters)
         if params == list() or func_arg_no == 0:
-            func()
+            callback()
         else:
-            func(params)
+            callback(params)
```

Comment:
these lines aren't covered by tests - maybe you could quickly add one where the callback is actually called?